pkg/utils/etcdutil/etcdutil.go

```diff
@@ -119,12 +120,17 @@ func ListEtcdMembers(ctx context.Context, client *clientv3.Client) (*clientv3.Me
 		time.Sleep(time.Duration(d) * time.Second)
 	})
 	newCtx, cancel := context.WithTimeout(ctx, DefaultRequestTimeout)
-	listResp, err := client.MemberList(newCtx)
+	// After the etcd server is upgraded to v3.5.0, the MemberList API will return the member list in a linearizable way by default.
+	// It is introduced by https://github.com/etcd-io/etcd/pull/11639
+	// If Linearizable is set to false, the member list will be returned with server's local data.
+	// If Linearizable is set to true, it is served with linearizable guarantee. If the server is disconnected from quorum, `MemberList` call will fail.
+	c := clientv3.RetryClusterClient(client)
+	resp, err := c.MemberList(newCtx, &etcdserverpb.MemberListRequest{Linearizable: false})
 	cancel()
 	if err != nil {
-		return listResp, errs.ErrEtcdMemberList.Wrap(err).GenWithStackByCause()
+		return (*clientv3.MemberListResponse)(resp), errs.ErrEtcdMemberList.Wrap(err).GenWithStackByCause()
```

Comment:
why do we need (*clientv3.MemberListResponse)?

Reply:
Because the return value of ListEtcdMembers is expected to be (*clientv3.MemberListResponse)

Reply:
What is the current type without (*clientv3.MemberListResponse)?

Reply:
Oh, I notice you are using retryClusterClient. Is it necessary?

Reply:
Current type is `etcdserverpb.MemberListResponse`

Reply:
In order to pass `&etcdserverpb.MemberListRequest{Linearizable: false}`

If we use `resp, err :=client.MemberList(ctx)`, it will always be true

https://github.com/etcd-io/etcd/blame/9a5533382d84999e4e79642e1ec0f8bfa9b70ba8/client/v3/cluster.go#L125-132

Reply:
`client.MemberList` call `c.remote.MemberList`. `c.remote` is built by `RetryClusterClient` 
https://github.com/etcd-io/etcd/blob/v3.5.15/client/v3/cluster.go#L60-L67